### DISCUSSION TITLE
Education software linked to screen time/parental control

### DISCUSSION THREAD

**Discussion by loulafg** [elementary/switchboard-plug-parental-controls#202] [Ideas]
Hi, I am a unexpirienced (very junior) dev, basically know C/C++, and just learned Elementary OS has parental control with a screen time manager. I'd like to implement a system where the child could answer quizzes previously put in place by the parent, on a spaced repetition fashion, and then their performance on the quiz would be used to "buy" screen time. It would be more or less a gamefied screen time budget manager for reviewing school material. I have never even heard of Vala, nor have I programed anything in GTK, or with graphic UI for that matter, but I do believe it is feasable. Can someone point me in the way of how to do this in Elementary?

**Comment by danirabbit** [comment]
This is a cool idea but also very ambitious since it would require adding a new API for apps to be able to ask permission to manage screen time.

Probably the best place to start would be to create a new feature request for a Dbus API in the Portals project here: https://github.com/elementary/portals

At the moment we only have the option to allow usage during certain hours and not based on a duration of time. It looks like there is an open feature request for this, so that would definitely be something that would also be a pre-requisite to implement: https://github.com/elementary/switchboard-plug-parental-controls/issues/47

**Reply by loulafg** [reply]
Thanks a lot for the input. I see there are tools not related to Elementary that I could try to expand uppon, and maybe I'll go that way. But do you believe that writing such API would be prohibitively complex for a beginner (maybe security or other system issues are involved)? I never took part in any open souce collaboration, so I might be making the most stupid questions here, but if I wanted to write the total time feature and latter the api, how exactly should I procede? Is there a specific way to get in touch with the devs working on the parental control or should I just shuffle aroud the code and find out how it works?

**Reply by danirabbit** [reply]
I'm not sure about prohibitively complex, but it's definitely not something I would consider a super beginner friendly endeavor. It might take some time as you'll likely go through a few rounds of feedback on it. Probably @Marukesu would have the best advice about how to go about the API

As for the total time feature, yeah I would recommend just diving into the codebase and see what you can see. Looks like maybe @davidmhewitt would be a person to ask for advice about where to start in there

Good luck and thanks for your interest in working on this!

**Reply by loulafg** [reply]
Thanks a lot, I'll dive in and see what I can get.

**Reply by davidjenner** [reply]
@Marukesu @davidmhewitt @loulafg
Any advance on this? Look like a neat idea. Reminds me of when Gmail had the maths test to prevent drunken messages at 2am! 😂
